indico/MaKaC/webinterface/rh/authorDisplay.py

```diff
@@ -14,25 +14,40 @@
 # You should have received a copy of the GNU General Public License
 # along with Indico; if not, see <http://www.gnu.org/licenses/>.
 
+from MaKaC.webinterface.displayMgr import ConfDisplayMgrRegistery
+
 from MaKaC.webinterface.pages import authors
 from MaKaC.webinterface import urlHandlers
 from MaKaC.webinterface.rh.contribDisplay import RHContributionDisplayBase
-from MaKaC.errors import MaKaCError
+from MaKaC.errors import MaKaCError, AccessError
+
+
+class RHAuthorDisplayBase(RHContributionDisplayBase):
 
-class RHAuthorDisplayBase( RHContributionDisplayBase ):
+    _MENU_ITEMS_TO_DISABLE = ('speakerIndex', 'authorIndex', 'contributionList')
 
-    def _checkParams( self, params ):
-        RHContributionDisplayBase._checkParams( self, params )
-        self._authorId = params.get( "authorId", "" ).strip()
+    def _checkParams(self, params):
+        RHContributionDisplayBase._checkParams(self, params)
+        self._authorId = params.get("authorId", "").strip()
         if self._authorId == "":
             raise MaKaCError(_("Author ID not found. The URL you are trying to access might be wrong."))
 
+    def _checkProtection(self):
+        RHContributionDisplayBase._checkProtection(self)
+        conference_menu = ConfDisplayMgrRegistery().getDisplayMgr(self._conf).getMenu()
+
+        for menu_item_to_disable in self._MENU_ITEMS_TO_DISABLE:
+            if not conference_menu.getLinkByName(menu_item_to_disable).isEnabled():
+                raise AccessError()
+            else:
+                return
```

Comment:
why? you don't have any code later so you can simply omit the whole `else: return` part